draft-ietf-jsonpath-base.md

```diff
@@ -1551,23 +1566,20 @@ string matches a given regular expression, which is in {{-iregexp}} form.
 $[?match(@.date, "1974-05-..")]
 ~~~
 
-Its first argument is an optional string that is matched against the iregexp
+Its first argument is a string that is matched against the iregexp
 contained in the string that is the second argument.
-The result is `true` if the string matches the iregexp and `false`
-otherwise.
-
-The result is `Nothing` if the first argument is not a string or
-the second argument is not a string conforming to {{-iregexp}}.
+The result is `BooleanTrue` if the string matches the iregexp
+and `BooleanFalse` otherwise.
```

Comment:
This doesn't say what happens if the first argument is not a string or if the second argument does not conform to i-regexp.

Reply:
"otherwise" covers those cases.